flake.nix

```diff
@@ -176,9 +176,11 @@
           export LIBSSH2_SYS_USE_PKG_CONFIG=1
 
           export RUSTFLAGS="-Zthreads=0"
-        '' + pkgs.lib.optionalString useMoldLinker ''
+        '' + (if pkgs.stdenv.isLinux then ''
           export RUSTFLAGS+=" -C link-arg=-fuse-ld=mold -C link-arg=-Wl,--compress-debug-sections=zstd"
-        '' + darwinNextestHack;
+        '' else ''
+          export RUSTFLAGS+=" -C link-arg=-fuse-ld=/usr/bin/ld -C link-arg=-ld_new"
```

Comment:
It would be nice to have a comment here explaining the motivation (linker performance) and that we need `/usr/bin/ld` to run the Xcode/Command Line Tools `ld(1)` because Nix’s doesn’t have the new linker.